cs/Compiler.csproj

```diff
@@ -0,0 +1,96 @@
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Templates.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\ApplyOverloads.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Apply_cpp.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Apply_h.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Comm_cpp.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Comm_h.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Enum_h.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Grpc_cpp.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Grpc_h.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Reflection_h.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Types_cpp.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Types_h.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Util.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cs\Comm_cs.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cs\Grpc_cs.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cs\Types_cs.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cs\Util.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cs\Grpc_cs.hs" />
+  </ItemGroup>
+
+  <Import Project="$(MSBuildThisFileDirectory)\build\internal\Common.Internal.targets" />
+
+  <PropertyGroup>
+    <ExternalBuildDependsOn>
+      CoreExternalBuild;
+    </ExternalBuildDependsOn>
+  </PropertyGroup>
+
+  <Choose>
+    <When Condition="'$(OS)' == 'Windows_NT'">
+      <PropertyGroup>
+        <BuildDependsOn>
+          ExternalBuild;
+        </BuildDependsOn>
+      </PropertyGroup>
+    </When>
+    <Otherwise>
+      <PropertyGroup>
+        <BuildDependsOn>
+          SkipExternalBuild;
+        </BuildDependsOn>
+      </PropertyGroup>
+    </Otherwise>
+  </Choose>
+
+  <Target Name="CoreExternalBuild"
+          Inputs="$(ProjectDir)\..\compiler\CMakeLists.txt;@(None)"
+          Outputs="$(ProjectDir)\tools\gbc.exe">
+    <Message Text="Building Bond Compiler..." Importance="High" />
+
+    <MakeDir Directories="$(IntermediateOutputPath)" />
+
+    <Exec Command="cmake $(ProjectDir)\..\compiler -Wno-dev" WorkingDirectory="$(IntermediateOutputPath)" />
+    <Exec Command="cmake --build . --target gbc" WorkingDirectory="$(IntermediateOutputPath)" />
+    
+    <Copy SourceFiles="$(IntermediateOutputPath)\build\gbc\gbc.exe" DestinationFolder="$(ProjectDir)\tools\" />
+  </Target>
+
+  <Target Name="ExternalBuild" DependsOnTargets="$(ExternalBuildDependsOn)" />
```

Comment:
It looks like ExternalBuild and CoreExternalBuild can be merged together, reducing the number of Targets. Any reason to keep them split?